src/services/organizeImports.ts

```diff
@@ -21,15 +21,23 @@ namespace ts.OrganizeImports {
         const topLevelImportDecls = sourceFile.statements.filter(isImportDeclaration);
         organizeImportsWorker(topLevelImportDecls);
 
+        // All of the old ExportDeclarations in the file, in syntactic order.
+        const topLevelExportDecls = sourceFile.statements.filter(isExportDeclaration);
+        organizeImportsWorker(topLevelExportDecls);
+
         for (const ambientModule of sourceFile.statements.filter(isAmbientModule)) {
             const ambientModuleBody = getModuleBlock(ambientModule as ModuleDeclaration);
+
             const ambientModuleImportDecls = ambientModuleBody.statements.filter(isImportDeclaration);
             organizeImportsWorker(ambientModuleImportDecls);
+
+            const ambientModuleExportDecls = ambientModuleBody.statements.filter(isExportDeclaration);
+            organizeImportsWorker(ambientModuleExportDecls);
         }
 
         return changeTracker.getChanges();
 
-        function organizeImportsWorker(oldImportDecls: ReadonlyArray<ImportDeclaration>) {
+        function organizeImportsWorker(oldImportDecls: ReadonlyArray<ImportDeclaration | ExportDeclaration>) {

```

Comment:
To keep it type-safe this could be written as:
```ts
function organizeImportsWorker<T extends ImportDeclaration | ExportDeclaration>(oldImportDecls: ReadonlyArray<T>, coalesce: (t: ReadonlyArray<T>) => ReadonlyArray<T>)
```
And called with:
```ts
const coalesceAndOrganizeImports = (importGroup: ReadonlyArray<ImportDeclaration>) => coalesceImports(removeUnusedImports(importGroup, sourceFile, program));
...
organizeImportsWorker(topLevelImportDecls, coalesceAndOrganizeImports);
...
organizeImportsWorker(topLevelExportDecls, coalesceExports);
```

Then inside the function you can avoid the casts and just:
```ts
const newImportDecls = flatMap(sortedImportGroups, importGroup => getExternalModuleName(importGroup[0].moduleSpecifier) ? coalesce(importGroup) : importGroup);
```

Reply:
Personally, I find that pulling a single line out of a method and moving it offscreen makes the method harder to read, but I agree that runtime type checks are undesirable (even though we're using JS).

Having said that, I found that I was fighting the type system for most of this change.  For instance, I would have liked to type the parameter as `ReadonlyArray<ImportDeclaration> | ReadonlyArray<ExportDeclaration>`, but that caused even more problems.

Reply:
I've implemented it and it looks reasonable.